erica/erica_legacy/pyeric/utils.py

```diff
@@ -1,12 +1,45 @@
 import os
+from functools import lru_cache
 
-from xml.dom import minidom
+from erica.erica_legacy.elster_xml import elster_xml_generator
+from erica.erica_legacy.elster_xml.xml_parsing.erica_xml_parsing import remove_declaration_and_namespace
+from erica.erica_legacy.pyeric.pyeric_controller import PermitListingPyericProcessController
 
 _INSTANCES_FOLDER = os.path.join('erica', 'instances')
 _BLUEPRINT_FOLDER = os.path.join(_INSTANCES_FOLDER, 'blueprint')
 
+NEW_REQUEST_ID_SINCE_LAST_CACHE_INVALIDATION = []
 
-def pretty_xml_string(xml_string):
-    dom_string = minidom.parseString(xml_string).toprettyxml(indent=" " * 4)
-    return '\n'.join([s for s in dom_string.splitlines() if s.strip()])  # remove empty lines
 
+def reset_new_request_id_list():
+    global NEW_REQUEST_ID_SINCE_LAST_CACHE_INVALIDATION
+    NEW_REQUEST_ID_SINCE_LAST_CACHE_INVALIDATION = []
+
+
+def add_new_request_id_to_cache_list(request_id):
+    global NEW_REQUEST_ID_SINCE_LAST_CACHE_INVALIDATION
+    NEW_REQUEST_ID_SINCE_LAST_CACHE_INVALIDATION.append(request_id)
```

Comment:
Yes, that was exactly the question I raised above. I'll have a look into how to test that properly.

Reply:
Any ideas would be very welcome

Reply:
I think I'd add the following test cases for `request_needs_testmerker(request_id)`:
- idnr in list -> cache invalidated + `get_vast_list_from_xml` called
- idnr not in list + no cache -> `get_vast_list_from_xml` called
- idnr not in list + cache -> `get_vast_list_from_xml` not called
So very similar to what you already have just with the idnr-in-list parameter^^

And for the FSC Request RequestController:
- list is updated

Reply:
> And for the FSC Request RequestController:
> list is updated

We test that by checking that the right function is called (that is the test I named incorrectly). We can also check for the list but this will only test the implementation of the add_to_cache function. wdyt?

Reply:
I have added the following tests:
- idnr in list -> cache invalidate called
- idnr not in list -> cache invalidate not called

I did not add the other tests because 1) there is no spyOn in pytest as far as I know and the mock would replace the cached function 2) we would only test the lru_cache implementation.

wdyt?

Reply:
> We test that by checking that the right function is called (that is the test I named incorrectly). We can also check for the list but this will only test the implementation of the add_to_cache function. wdyt?

Just testing for the function call should be fine. 👍 

> wdyt?

Great.